tests/modules/saml/src/Auth/Source/SPTest.php

```diff
@@ -482,6 +482,44 @@ public function testSPIdpListScoping(): void
         );
     }
 
+    /**
+     * Test that adding IDPList to the idp metadata
+     * will result in that IDP being added to the scope
+     */
+    public function testIdpMetadataScoping(): void
+    {
+        $this->idpConfigArray['IDPList'] = ['https://scope.example.com'];
+        $ar = $this->createAuthnRequest([]);
+
+        $this->assertTrue(
+            in_array('https://scope.example.com', $ar->getIDPList())
+        );
+    }
+
+    /**
+     * Test that adding IDPList to the config
+     * will result in that IDP being added to the scope
+     */
+    public function testRemoteMetadataScoping(): void
+    {
+        $info = ['AuthId' => 'default-sp'];
+        $config = [
+            'IDPList' => ['https://scope.example.com']
+        ];
+        $as = new SpTester($info, $config);
+
+        /** @var \SAML2\AuthnRequest $ar */
+        try {
+            $as->startSSO2Test($this->getIdpMetadata(), []);
+            $this->assertTrue(false, 'Expected ExitTestException');
+        } catch (ExitTestException $e) {
+            ['ar' => $ar] = $e->getTestResult();
+            $this->assertTrue(
+                in_array('https://scope.example.com', $ar->getIDPList())
+            );
```

Comment:
This can be written as `$this->assertContains('https://scope.example.com', $ar->getIDPList());`